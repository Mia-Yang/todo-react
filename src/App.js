import './App.css';
import TodoList from './component/TodoList';
import TodoHeader from './component/TodoHeader'


function App() {
  return (
    <main>
        <TodoHeader />
        <TodoList />
    </main>

  );
}

export default App;
