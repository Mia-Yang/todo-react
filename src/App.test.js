import { fireEvent, getByRole, render, screen } from '@testing-library/react';
import App from './App';

test('should take a snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});

test('should capture changes in inputbox', () => {
  const app = render(<App />);
  const inputElement = app.getByTestId('inputbox');
  fireEvent.change(inputElement, { target: { value: 'workout' } });
  expect(inputElement.value).toBe('workout');
});

test('should create a new todo', () => {
  const app = render(<App />);

  const inputElement = app.getByTestId('inputbox');
  const createButtonElement = app.getByTestId('addTodoButton');

  // Create the todo.
  fireEvent.change(inputElement, { target: { value: 'Feed my dog.' } });
  fireEvent.click(createButtonElement);

  expect(app.getByTestId('todo')).toHaveTextContent('Feed my dog');

  expect(inputElement.value).toBe('');

  expect(app.getByTestId('todo')).toBeInTheDocument();

  expect(app.getAllByTestId('todo').length).toBe(1);
});

test('should be checked when the item is done', () => {
  const app = render(<App />);
  const inputElement = app.getByTestId('inputbox');
  const createButtonElement = app.getByTestId('addTodoButton');

  // Create the todo.
  fireEvent.change(inputElement, { target: { value: 'workout 30mins.' } });
  fireEvent.click(createButtonElement);
  const checkbox = app.getByRole('checkbox');
  fireEvent.click(checkbox);
  expect(checkbox).toBeChecked;
});

test('should delete item when click its delete button', () => {
  const app = render(<App />);

  const inputElement = app.getByTestId('inputbox');
  const createButtonElement = app.getByTestId('addTodoButton');

  expect(app.queryAllByTestId('todo').length).toBe(0);

  // Create the todo.
  fireEvent.change(inputElement, { target: { value: 'workout 30mins.' } });
  fireEvent.click(createButtonElement);

  expect(app.queryAllByTestId('todo').length).toBe(1);

  // Click the delete button
  const deleteButton = app.getByTestId('deleteTodo');
  fireEvent.click(deleteButton);

  expect(inputElement.value).toBe('');

  expect(app.queryByTestId('todo')).not.toBeInTheDocument();

  expect(app.queryAllByTestId('todo').length).toBe(0);
});

test('should clear all items when click clear button', () => {
  const app = render(<App />);

  const inputElement = app.getByTestId('inputbox');
  const createButtonElement = app.getByTestId('addTodoButton');

  // Create the todo.
  fireEvent.change(inputElement, { target: { value: 'workout 30mins.' } });
  fireEvent.click(createButtonElement);
  fireEvent.change(inputElement, { target: { value: 'water flowers.' } });
  fireEvent.click(createButtonElement);

  expect(app.queryAllByTestId('todo').length).toBe(2);

  // Click the delete button
  const clearButton = app.getByTestId('clearButton');
  fireEvent.click(clearButton);

  expect(inputElement.value).toBe('');

  expect(app.queryAllByTestId('todo').length).toBe(0);
});
