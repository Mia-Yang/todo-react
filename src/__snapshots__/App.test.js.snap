// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <main>
    <header>
      <h1>
        Todos
      </h1>
    </header>
    <div>
      <div>
        <form>
          <input
            data-testid="inputbox"
            placeholder="Plan your life"
            type="text"
            value=""
          />
          <button
            class="add-btn"
            data-testid="addTodoButton"
          >
            âž•
          </button>
        </form>
        <button
          class="clear"
          data-testid="clearButton"
        >
          Clear All
        </button>
      </div>
      <ul
        class="todo-list"
      />
    </div>
  </main>
</DocumentFragment>
`;
